src/MLJXGBoostInterface.jl

```diff
@@ -1,770 +1,212 @@
-`Binary=Finite{2}`). Gives probabilistic predictions. For admissible
-`kwargs`, see
-[https://xgboost.readthedocs.io/en/latest/parameter.html](https://xgboost.readthedocs.io/en/latest/parameter.html).
-
-For a time-dependent random seed, use `seed=-1`.
-
-See also: XGBoostCount, XGBoostRegressor
-
-"""
-function XGBoostClassifier(
-    ;num_round=100
-    ,booster="gbtree"
-    ,disable_default_eval_metric=0
-    ,eta=0.3
-    ,gamma=0
-    ,max_depth=6
-    ,min_child_weight=1
-    ,max_delta_step=0
-    ,subsample=1
-    ,colsample_bytree=1
-    ,colsample_bylevel=1
-    ,lambda=1
-    ,alpha=0
-    ,tree_method="auto"
-    ,sketch_eps=0.03
-    ,scale_pos_weight=1
-    ,updater="auto"
-    ,refresh_leaf=1
-    ,process_type="default"
-    ,grow_policy="depthwise"
-    ,max_leaves=0
-    ,max_bin=256
-    ,predictor="cpu_predictor" #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type="uniform"
-    ,normalize_type="tree"
-    ,rate_drop=0.0
-    ,one_drop=0
-    ,skip_drop=0.0
-    ,feature_selector="cyclic"
-    ,top_k=0
-    ,tweedie_variance_power=1.5
-    ,objective="automatic"
-    ,base_score=0.5
-    ,eval_metric="mlogloss"
-    ,seed=0
-    ,nthread = 1)
-
-    model = XGBoostClassifier(
-    num_round
-    ,booster
-    ,disable_default_eval_metric
-    ,eta
-    ,gamma
-    ,max_depth
-    ,min_child_weight
-    ,max_delta_step
-    ,subsample
-    ,colsample_bytree
-    ,colsample_bylevel
-    ,lambda
-    ,alpha
-    ,tree_method
-    ,sketch_eps
-    ,scale_pos_weight
-    ,updater
-    ,refresh_leaf
-    ,process_type
-    ,grow_policy
-    ,max_leaves
-    ,max_bin
-    ,predictor #> gpu version not currently working with Julia, maybe remove completely?
-    ,sample_type
-    ,normalize_type
-    ,rate_drop
-    ,one_drop
-    ,skip_drop
-    ,feature_selector
-    ,top_k
-    ,tweedie_variance_power
-    ,objective
-    ,base_score
-    ,eval_metric
-    ,seed
-    ,nthread)
-
-     message = MMI.clean!(model)           #> future proof by including these
-     isempty(message) || @warn message #> two lines even if no clean! defined below
-
-    return model
-end
+eval(modelexpr(:XGBoostCount, :XGBoostAbstractRegressor, "count:poisson", :validate_count_objective))
 
 
-function MMI.clean!(model::XGBoostClassifier)
-    warning = ""
-    if(!(model.objective =="automatic"))
-            warning *="Changing objective to \"automatic\", the only supported value. "
-            model.objective="automatic"
-    end
-    return warning
-end
+eval(modelexpr(:XGBoostClassifier, :XGBoostAbstractClassifier, "automatic", :validate_class_objective))
 
 function MMI.fit(model::XGBoostClassifier
-                     , verbosity::Int     #> must be here even if unsupported in pkg
-                     , X
-                     , y)
-    Xmatrix = _to_array(MMI.matrix(X))
-
+                 , verbosity::Int     #> must be here even if unsupported in pkg
```

Comment:
```suggestion
                 , verbosity
```

Typing `verbosity` is now an anachronism and not recommended.

Reply:
Note that there is an implicit assumption in here that it's an `Integer`, or at least `Real` because it is compared to `0`.  I still took out the type assertions in case there are any dispatch worries.